rumqttc/src/eventloop.rs

```diff
@@ -119,11 +119,12 @@ impl EventLoop {
         }
     }
 
-    fn clean(&mut self) {
+    /// Last session might contain packets which aren't acked. MQTT says these packets should be
+    /// republished in the next session. Move pending messages from state to eventloop.
+    pub fn clean(&mut self) {
```

Comment:
the docs comments on this fn should also mention it would drop the network and keepalive, and thus disconnect the connection with broker ( but won't  send disconnect packet like `client.disconnect()` does )

PS: would be nice if we can specify when we should/ shouldn't use this instead of disconnect ( not that imp though )